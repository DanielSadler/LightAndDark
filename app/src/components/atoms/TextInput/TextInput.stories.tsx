import { storiesOf } from "@storybook/react-native";
import React from "react";

import { TextInput } from "./TextInput";

storiesOf("Atoms/TextInput", module)
  .add("label", () => (
    <TextInput label="Email Address" keyboardType="email-address" />
  ))
  .add("placeholder", () => (
    <TextInput placeholder="Email Address" keyboardType="email-address" />
  ));
