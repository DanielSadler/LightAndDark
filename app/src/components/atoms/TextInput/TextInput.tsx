import { ComponentProps } from "react";
import { Text, TextInput as RNTextInput, View } from "react-native";

type Props = ComponentProps<typeof RNTextInput> & { label?: string };

export const TextInput = ({ label, className, ...props }: Props) => (
  <View className={className}>
    {label ? (
      <Text className="block text-sm font-medium text-gray-700 my-1">
        {label}
      </Text>
    ) : (
      <View />
    )}
    <RNTextInput
      className="appearance-none rounded-md border border-gray-300 px-3 py-2 placeholder-black shadow-sm focus:border-indigo-500 focus:outline-none focus:ring-indigo-500 sm:text-sm"
      {...props}
    />
  </View>
);
