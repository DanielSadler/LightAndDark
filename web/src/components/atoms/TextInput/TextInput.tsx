import React, { FC } from 'react';

type Props = React.DetailedHTMLProps<
  React.InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
> & {
  label: string;
};

export const TextInput: FC<Props> = ({ label, ...props }) => (
  <div>
    <label htmlFor="email" className="block text-sm font-medium text-gray-700">
      {label}
    </label>
    <div className="mt-1">
      <input
        {...props}
        className="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
      />
    </div>
  </div>
);
