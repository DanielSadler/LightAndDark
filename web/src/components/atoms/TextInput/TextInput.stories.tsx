import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { TextInput } from './TextInput';

export default {
  title: 'Atoms/TextInput',
  component: TextInput,
} as ComponentMeta<typeof TextInput>;

const Template: ComponentStory<typeof TextInput> = (args) => (
  <TextInput {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  label: 'Email address',
  id: 'email',
  name: 'email',
  type: 'email',
  autoComplete: 'email',
};
export const Password = Template.bind({});
Password.args = {
  label: 'Password',
  id: 'password',
  name: 'password',
  type: 'password',
  autoComplete: 'current-password',
};
