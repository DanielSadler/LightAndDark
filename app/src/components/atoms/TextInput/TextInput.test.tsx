import * as React from "react";
import * as renderer from "react-test-renderer";

import { TextInput } from "./TextInput";

describe("TextInput", () => {
  it("should match a snapshot", () => {
    const tree = renderer.create(<TextInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
