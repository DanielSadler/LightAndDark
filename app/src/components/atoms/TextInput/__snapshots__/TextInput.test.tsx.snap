// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should match a snapshot 1`] = `
<View>
  <View />
  <TextInput
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      [
        {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
        {
          "borderBottomWidth": 1,
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
        },
        {
          "borderBottomColor": "#d1d5db",
          "borderLeftColor": "#d1d5db",
          "borderRightColor": "#d1d5db",
          "borderTopColor": "#d1d5db",
        },
        {
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        {
          "paddingBottom": 8,
          "paddingTop": 8,
        },
        {
          "color": "#000",
        },
        {
          "shadowColor": "rgba(0, 0, 0, 0.1)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 2,
        },
        {
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  />
</View>
`;
